services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -814,7 +814,7 @@ protected Response handleLogoutResponse(SAMLDocumentHolder holder, StatusRespons
 
         private String getExpectedDestination(String providerAlias, String clientId) {
             if(clientId != null) {
-                return session.getContext().getUri().getAbsolutePath().toString();
+                return Urls.identityProviderAuthnResponse(session.getContext().getUri().getBaseUri(), providerAlias, realm.getName()).toString()+"/clients/"+clientId;
```

Comment:
Rather than building the path here manually I'd say let's add an overload to the `Urls.identityProviderAuthnResponse()` that accepts a client so that the `URIBuilder` can be re-used.

Reply:
I have updated the PR with this change